src/sdk/proxy-callbacks.ts

```diff
@@ -0,0 +1,104 @@
+import { Context, SupportedEventsU } from "../sdk/context";
+import { CallbackFunction, ProxyCallbacks } from "../types/helpers";
+import { postWorkerErrorComment } from "./errors";
+
+/**
+ * Build your callbacks first and pass `CallbackBuilder` directly to `createPlugin`.
+ * 
+ * @example
+
+```ts
+const builder = new CallbackBuilder()
+  .addCallback("issue_comment.created", <CallbackFunction<"issue_comment.created">>helloWorld)
+  .addCallback("issue_comment.deleted", <CallbackFunction<"issue_comment.deleted">>goodbyeCruelWorld);
+   ```
+ */
+export class CallbackBuilder {
+  private _callbacks: ProxyCallbacks = {} as ProxyCallbacks;
+
+  /**
+   * Add a callback for the given event.
+   *
+   * @param event The event to add a callback for.
+   * @param callback The callback to add.
+   */
+  addCallback<TEvent extends SupportedEventsU>(event: TEvent, callback: CallbackFunction<TEvent>) {
+    this._callbacks[event] ??= [];
+    this._callbacks[event].push(callback);
+    return this;
+  }
+
+  /**
+   * This simply returns the callbacks object.
+   */
+  build() {
+    return this._callbacks;
+  }
+}
+
+export function proxyCallbacks(context: Context, callbackBuilder: CallbackBuilder): ProxyCallbacks {
+  return new Proxy(callbackBuilder.build(), {
```

Comment:
why do we need a Proxy though?

Reply:
Truthfully it was @gentlementlegen that first introduced using a Proxy like this he'd give a better answer I'd imagine. 

I assumed it was because it abstracts away some of the logic you'd typically write. I'm sure I said something like "looks like heavy firepower" lmao but with the Types built around it now it's clean in handling all webhook events, which is why I like it.

Similar could be achieved taking the array of callbacks directly from the obj and `Promise.all()` I guess

Reply:
It was to avoid a chaining of "if else if else if" at first because there started to be lots of events that had to be handled.

Reply:
I mean here we just have an object with event names as keys and array of callbacks to call, so there's no `if else if else` so I think Proxy just complicates things and removing it would make it simpler

Reply:
Definitely not having it makes it simpler. Again it was just because the plugin had reached a point where it became ugly. The use of proxy ensures that if a key that didn't exist was called on the object, an error message was returned instead, that's all.